lib/class-wp-rest-menu-items-controller.php

```diff
@@ -260,14 +294,16 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
+		// Check if object id existing before saving.
 		if ( ! $prepared_nav_item['menu-item-object'] && $prepared_nav_item['menu-item-object-id'] ) {
+			// If taxonony, check if term exists.
 			if ( 'taxonomy' === $prepared_nav_item['menu-item-type'] ) {
 				$original = get_term( (int) $prepared_nav_item['menu-item-object-id'] );
 				if ( empty( $original ) ) {
 					return new WP_Error( 'rest_term_invalid_id', __( 'Invalid term ID.' ), array( 'status' => 400 ) );
 				}
 				$prepared_nav_item['menu-item-object'] = get_term_field( 'taxonomy', $original );
-			} elseif ( 'post_type' === $prepared_nav_item['menu-item-type'] ) {
+			} elseif ( 'post_type' === $prepared_nav_item['menu-item-type'] ) { // If post, check if post object exists.
```

Comment:
small style thing: comments should be bumped to a line of their own in cases like this.